tao_compiler/mlir/xla/ral/context/common_context_impl_quantization.cc

```diff
@@ -159,10 +159,184 @@ void ral_qconv_s8_s8_s8(
+    std::shared_ptr<AclConvInfo> info(new AclConvInfo);
+    auto src_shape = TensorShape(Ci, Iw, Ih, N);
+    auto weights_shape = TensorShape(Ci, Kw, Kh, Co);
+    auto dst_shape = TensorShape(Co, Ow, Oh, N);
+
+    DataLayout data_layout = DataLayout::NHWC;
+    DataType data_type = DataType::QASYMM8_SIGNED;
+    TensorInfo src_info = TensorInfo(src_shape, 1, data_type, data_layout);
+    TensorInfo weights_info =
+        TensorInfo(weights_shape, 1, DataType::QSYMM8_PER_CHANNEL, data_layout);
+    TensorInfo dst_info = TensorInfo(dst_shape, 1, data_type, data_layout);
+
+    const QuantizationInfo src_qinfo = QuantizationInfo();
+    src_info.set_quantization_info(
+        QuantizationInfo(*inputScales.data, *inputZeroPoints.data));
+    std::vector<float> scales(weightScales.data,
+                              weightScales.data + weightScales.sizes[0]);
+    std::vector<int32_t> zero_points(
+        weightZeroPoints.data,
+        weightZeroPoints.data + weightZeroPoints.sizes[0]);
+    weights_info.set_quantization_info(
+        QuantizationInfo(std::move(scales), std::move(zero_points)));
+    dst_info.set_quantization_info(
+        QuantizationInfo(*resultScales.data, *resultZeroPoints.data));
+
+    info->src.allocator()->init(src_info);
+    info->weights.allocator()->init(weights_info);
+    info->dst.allocator()->init(dst_info);
+    info->src.allocator()->import_memory(input.data);
+    info->weights.allocator()->import_memory(weight.data);
+    info->dst.allocator()->import_memory(result.data);
+
+    if (!info->op.validate(
+            &src_info, &weights_info, nullptr, &dst_info,
+            PadStrideInfo{params.strides[1], params.strides[0],
+                          params.padding_l[1], params.padding_r[1],
+                          params.padding_l[0], params.padding_r[0],
+                          DimensionRoundingType::FLOOR},
+            WeightsInfo{}, Size2D{params.dilates[1], params.dilates[0]})) {
+      ctx->signalError(
+          Context::FAILURE,
+          "fail to validate ral_qconv_acl_s8_s8_s8_per_channel conv");
+    } else {
+      info->op.configure(&info->src, &info->weights, nullptr, &info->dst,
+                         PadStrideInfo{params.strides[1], params.strides[0],
+                                       params.padding_l[1], params.padding_r[1],
+                                       params.padding_l[0], params.padding_r[0],
+                                       DimensionRoundingType::FLOOR},
+                         WeightsInfo{},
+                         Size2D{params.dilates[1], params.dilates[0]});
+    }
+    if (pack) info->op.reuse_packed_weight(*pack);
+    info->op.prepare(&info->src, &info->weights, nullptr, &info->dst);
+    return info;
+  };
+
+  std::shared_ptr<AclConvInfo> info;
+  if (isWeightPrePackingEnabled() && params.weight_is_const) {
+    std::string unique_name = "disc.ral_qconv_acl_s8_s8_s8_per_channel";
```

Comment:
nit: any naming convention here? I saw `tao.cpu.blabla` before.

Reply:
better to use disc as prefix for new code?